### ISSUE TITLE
Blank homepage after Debian update

### ISSUE THREAD

**Issue by GTP95** [firefly-iii/firefly-iii#2748]
I am running Firefly III version 4.8.1.4

**Description**
I was able to setup Firefly III and get to the point where I could reach the registration page via browser, but didn't registered any account yet. I left it as it was for some days, then four days ago I updated some packages on my Debian installation and checked the home page again, but this time I got a blank page. I'm quite sure that this is not caused by my Apache config (did some tests). In firefly's log folder I found two files, the one with the date of that day is called "ff3-apache2handler-2019-10-15.log" and contains the following error

> [2019-10-15 13:05:22] local.ERROR: Class hash does not exist {"exception":"[object] (ReflectionException(code: -1): Class hash does not exist at /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php:788)
[stacktrace]
#0 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(788): ReflectionClass->__construct('hash')
#1 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(667): Illuminate\\Container\\Container->build('hash')
#2 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(615): Illuminate\\Container\\Container->resolve('hash', Array)
#3 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(767): Illuminate\\Container\\Container->make('hash', Array)
#4 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(1227): Illuminate\\Foundation\\Application->make('hash')
#5 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/CreatesUserProviders.php(82): Illuminate\\Container\\Container->offsetGet('hash')
#6 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/CreatesUserProviders.php(40): Illuminate\\Auth\\AuthManager->createEloquentProvider(Array)
#7 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(123): Illuminate\\Auth\\AuthManager->createUserProvider('users')
#8 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(94): Illuminate\\Auth\\AuthManager->createSessionDriver('web', Array)
#9 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(68): Illuminate\\Auth\\AuthManager->resolve('web')
#10 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(297): Illuminate\\Auth\\AuthManager->guard()
#11 /mnt/www/firefly-iii/app/Exceptions/Handler.php(132): Illuminate\\Auth\\AuthManager->__call('check', Array)
#12 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(314): FireflyIII\\Exceptions\\Handler->report(Object(ErrorException))
#13 /mnt/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(118): Illuminate\\Foundation\\Http\\Kernel->reportException(Object(ErrorException))
#14 /mnt/www/firefly-iii/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#15 {main}
"} 

repeated a few times. If you need it here is a pastebin of the full log: [https://pastebin.com/T2z8nzDu](url). My understanding is that I'm missing a library, but I don't know which one. 
Thanks for your help. 

**Extra info**
Although I think the Debian update s the most likely cause I can't be sure it was the update fault because:
1) I didn't check if it was working immidiately before the update
2) Due to my setup, the SBC that runs Firefly has suufered some power loss, I'm working to fix this.

**Bonus points**

**Comment by JC5** [comment]
Make sure your webserver is running PHP 7.3, and run `composer install` just in case. Is there still an .env file with a correct config?

**Comment by GTP95** [comment]
Thanks for your help. I have PHP 7.3.9 and the config in the .env file looks fine. I've ran `composer install` and during the Generating optimized autoload files" phase gave me the following error:

> RuntimeException]                                                                                  
  Could not scan for classes inside "database/seeds" which does not appear to be a file nor a folder

After this I tried to connect to see if this solved the problem anyway and I got a 404 error. After some investigation I found out that I don't have the "public" folder inside "firefly-iii". I don't know if my Apache config was right, but previously I got it working bi setting the document root to /var/www/firefly-iii/public but this doesn't work anymore. Should I reinstall firefly or there is something else I should try?
